### PR TITLE
feat: question subcomponents

### PR DESCRIPTION
**Pull request by kaljarv**
## WHY:

Creates a suite of components for consistent display of questions for
both voters and candidates as well as for answering and presentation
of already given answers.

- Needed for #109 and #198 
- Partially solves #280

### What has been changed (if possible, add screenshots, gifs, etc. )

Creates several components for displaying different parts of
a question, such as `QuestionActions`, which displays the additional
actions available, such as 'Next', 'Previous' and 'Delete answer'.

The purpose for providing these partial components is to show the
questions consistently for both voters and candidates when answering
because any differences in how they're displayed may effect answers.

Furthermore, the `LikertResponseButtons` now supports use both for
answering (for both the voter and a candidate) and for displaying
answers when a voter is looking at a candidate's profile.

The `/(voters)/questions` route is update to show the components in
action. They are also quickly added to the `CandidateOpinions`
component.

Furthermore, there is a preliminary update to the 
`/candidate/(protected)/questions` route because the `Question`
component is no longer available.

Finally, there are small updates to global types, `Expander` styling
and a `onKeyboardFocusOut` Svelte action.

## Check off each of the following tasks as they are completed

- [x] I have reviewed the changes myself in this PR. Please check the [self-review document](https://github.com/OpenVAA/voting-advice-application/blob/main/docs/contributing/self-review.md)
- [ ] I have added or edited unit tests.
- [ ] I have run the unit tests successfully.
- [x] I have tested this change on my own device.
- [ ] I have tested this change on other devices (Using Browserstack is recommended).
- [x] I have tested my changes using the [WAVE extension](https://wave.webaim.org/extension/)
- [x] I have added documentation where necessary.
- [x] Is there an existing issue linked to this PR?

**Clean up your git commit history before submitting the pull request!**

### CODE REVIEW

**Review by jaguarfi — COMMENTED** [review]
LGTM

File: frontend/src/lib/components/expander/Expander.svelte [hunk 1]
```diff
@@ -66,47 +60,43 @@
...
+
+  <Expander title="Example title" variant="category"  iconColor="primary" 
+    titleClass="bg-base-100 text-primary" contentClass="bg-base-300 text-info font-bold">
+    <p>Example content<p/>
+  </Expander>
+  ```
+-->
 
 <div class={collapseClasses}>
   <input type="checkbox" aria-label="open ${title}" on:click={toggleExpanded} />
```
Comment (jaguarfi) [inline frontend/src/lib/components/expander/Expander.svelte:102]: What about the `aria-label` when the expander is supposed to close? Maybe we need to update the label dynamically.

File: frontend/src/lib/components/questions/LikertResponseButtons.svelte [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+  options={question.options}
+  mode="display"
+  selectedKey={$voterAnswers[question.id]}
+  entityKey={candidateAnswer}
+  entityLabel={$_('candidate.candidateAnswerLabel')}
+   />
+```
+-->
+
+<fieldset
```
Comment (jaguarfi) [inline frontend/src/lib/components/questions/LikertResponseButtons.svelte:203]: If users want to use the keyboard, how should they navigate the answers? Will they use arrow keys or the Tab key?

File: frontend/src/lib/utils/onKeyboardFocusOut.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ *   has left the container. We listen to `focusout` and `focusin` to
+ *   track whether we have focus or not. (We cannot use
+ *   `document.activeElement` because it's not reliable.)
+ *
+ * @param node The target element
+ * @param callback The event handler to call
+ *
+ * @example `<Navigation on:keyboardFocusOut={closeDrawer} />`
+ */
+export function onKeyboardFocusOut(node: HTMLElement, callback: () => void) {
```
Comment (jaguarfi) [inline frontend/src/lib/utils/onKeyboardFocusOut.ts:23]: This looks a bit hacky. We can revisit this in the future and find other alternatives.

File: frontend/src/lib/components/expander/Expander.svelte (hunk 1 above)
Comment (kaljarv) [reply frontend/src/lib/components/expander/Expander.svelte:102]: We should actually get rid of the label altogether and use this [pattern](https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/examples/disclosure-faq/). We had a discussion about that in the Candidate App PR that introduced `Expander`, but it was left unresolved.

I made an issue out of it #341

File: frontend/src/lib/components/questions/LikertResponseButtons.svelte (hunk 2 above)
Comment (kaljarv) [reply frontend/src/lib/components/questions/LikertResponseButtons.svelte:203]: They will use the arrow keys, which is the default way a radio group operates. `Tab` will move focus out of the widget, which is also the default.

File: frontend/src/lib/utils/onKeyboardFocusOut.ts (hunk 3 above)
Comment (kaljarv) [reply frontend/src/lib/utils/onKeyboardFocusOut.ts:23]: I agree, it's quite contrived. I spent quite some time trying come up with a nicer solution, but this was the only working one I could find. One tricky issue is that when tabbing through the navigation menu, keyboard focus will move out of the page after you reach the last nav item. Thus, we won't have any focused element on the page anymore…

**Pull request merged**